core/src/main/java/org/apache/iceberg/LocationProviders.java

```diff
@@ -104,9 +104,11 @@ public String newDataLocation(String filename) {
   }
 
   static class ObjectStoreLocationProvider implements LocationProvider {
-    private static final Function<Object, Integer> HASH_FUNC =
-        Transforms.bucket(Integer.MAX_VALUE).bind(Types.StringType.get());
 
+    private static final HashFunction HASH_FUNC = Hashing.sha1();
+
+    private static final String allChars =
+        "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
```

Comment:
nit: space between static and instance variable. `allChars` need to be capitalized, also can we add a length variable `NUM_CHARS=62`?

Reply:
`static final` should be `ALL_CHARS` right?